drivers/ipm/ipm_stm32_ipcc.c

```diff
@@ -88,7 +88,9 @@ LOG_MODULE_REGISTER(ipm_stm32_ipcc, CONFIG_IPM_LOG_LEVEL);
 struct stm32_ipcc_mailbox_config {
 	void (*irq_config_func)(const struct device *dev);
 	IPCC_TypeDef *ipcc;
+#if DT_INST_NUM_CLOCKS(0) != 0
```

Comment:
Nit: prefer
```suggestion
#if DT_INST_NUM_CLOCKS(0) > 0
```

Ditto all instances in this file.